api/base/serializers.py

```diff
@@ -34,6 +36,59 @@ def _url_val(val, obj, serializer, **kwargs):
         return val
 
 
+class HyperlinkedIdentityFieldWithMeta(ser.HyperlinkedIdentityField):
+    """
+    HyperlinkedIdentityField that returns a nested dict with url,
+    optional meta information, and link_type.
+
+    Example:
+
+        children = HyperlinkedIdentityFieldWithMeta(view_name='nodes:node-children', lookup_field='pk',
+                                    link_type='related', lookup_url_kwarg='node_id', meta={'count': 'get_node_count'})
+
+    """
+
+    def __init__(self, view_name=None, **kwargs):
+        kwargs['read_only'] = True
+        kwargs['source'] = '*'
+        self.meta = kwargs.pop('meta', None)
+        self.link_type = kwargs.pop('link_type', 'url')
+        super(ser.HyperlinkedIdentityField, self).__init__(view_name, **kwargs)
+
+    # overrides HyperlinkedIdentityField
+    def get_url(self, obj, view_name, request, format):
+        """
+        Given an object, return the URL that hyperlinks to the object.
+
+        Returns null if lookup value is None
+        """
+
+        if getattr(obj, self.lookup_field) is None:
+            return None
+
+        return super(ser.HyperlinkedIdentityField, self).get_url(obj, view_name, request, format)
+
+    # overrides HyperlinkedIdentityField
+    def to_representation(self, value):
+        """
+        Returns nested dictionary in format {'links': {'self.link_type': ... }
+
+        If no meta information, self.link_type is equal to a string containing link's URL.  Otherwise,
+        the link is represented as a links object with 'href' and 'meta' members.
+        """
+        url = super(HyperlinkedIdentityFieldWithMeta, self).to_representation(value)
+
+        if self.meta is None:
```

Comment:
Better to handle any falsy value here:

```python
if not self.meta:
```

This will, for example, handle an empty dict properly.